crates/release_plz_core/src/registry_packages.rs

```diff
@@ -62,17 +63,30 @@ pub fn get_registry_packages(
         ),
         None => {
             let temp_dir = tempdir().context("failed to get a temporary directory")?;
-            let local_packages_names: Vec<&str> =
-                local_packages.iter().map(|c| c.name.as_str()).collect();
             let directory = temp_dir.as_ref().to_str().context("invalid tempdir path")?;
 
-            let mut downloader = download::PackageDownloader::new(local_packages_names, directory);
-            if let Some(registry) = registry {
-                downloader = downloader.with_registry(registry.to_string());
+            // download packages from the registry defined in cli and fallback to the Cargo.toml `publish` field.
+            // HACK use the first registry in the `publish`
+            let mut registry_packages: Vec<Package> = vec![];
+            for (registry, packages) in &local_packages.iter().chunk_by(|p| {
+                registry.or_else(|| {
+                    p.publish
+                        .as_ref()
+                        .and_then(|p| p.first())
+                        .map(|x| x.as_str())
+                })
+            }) {
```

Comment:
can you extract what you are iterating on into a function or variable?
Maybe it's because it's quite late here, but I don't understand the use of chunk_by.

Maybe assigning a name to it and some comments will help